modules/geo-layers/src/mesh-layer/mesh-layer-vertex.glsl.js

```diff
@@ -26,9 +26,17 @@ out vec3 normals_commonspace;
 out vec4 position_commonspace;
 out vec4 vColor;
 
+vec2 applyUVRegion(vec2 uv) {
+  #ifdef HAS_UV_REGIONS
+    // https://github.com/Esri/i3s-spec/blob/master/docs/1.7/geometryUVRegion.cmn.md
+    return fract(uv) * (uvRegions.zw - uvRegions.xy) + uvRegions.xy;
+  #else
+    return uv;
+  #endif
+}
+
 void main(void) {
-  // https://github.com/Esri/i3s-spec/blob/master/docs/1.7/geometryUVRegion.cmn.md
-  vec2 uv = fract(texCoords) * (uvRegions.zw - uvRegions.xy) + uvRegions.xy;
+  vec2 uv = applyUVRegion(texCoords);
```

Comment:
Is it calling the `fract` that is problematic? I suppose we could also have added a uniform...

Reply:
What for? `fract` cuts meaning data. We should only call it if `uvRegions` is present in the tile content.

Reply:
My point was that the alternative is to dynamically disable code with uniforms, then you don't have to recompile the shader when changing behavior or risk that one branch of the shader breaks because it is not being tested etc.

#ifdef is a quite intrusive mechanism.

Reply:
But no need to change, this is fine. Just wanted to make sure my point was understood.

Reply:
I see. Thank you